network/ieee802_11_prism.ksy

```diff
@@ -0,0 +1,113 @@
+meta:
+  id: ieee802_11_prism
+  title: prism II headers for 802.11 capture
+  license: Unlicense
+  endian: le
+  encoding: ascii
+  xref:
+    wikidata: Q7245672
+doc: |
+  Prism II headers are usually used by wlan cards based on Intersil / Conexant [Prism II](https://en.wikipedia.org/wiki/Prism_(chipset)).
+
+doc-ref: https://www.tcpdump.org/linktypes/LINKTYPE_IEEE802_11_PRISM.html
+
+seq:
+  - id: message_code
+    type: u4
+  - id: size
+    type: u4
+  - id: device_name
+    type: strz
+    size: 16
+  - id: fields
+    type: fields
+    size: fields_size
+  - id: body
+    size-eos: true
+instances:
+  header_size:
+    value: 4 + 4 + 16
+  fields_size:
+    value: size - header_size
+types:
+  fields:
+    seq:
+      - id: fields
+        type: field
+        repeat: eos
+    types:
+      field:
+        seq:
+          - id: header
+            type: header
+          - id: is_not_supplied
+            type: u2
+          - id: size
+            type: u2
+          - id: value
+            size: size
+            type:
+              switch-on: header.type
+              cases:
+                'type::host_time': u4
+                'type::mac_time': u4
+                'type::channel': u4
+                'type::rssi': u4
+                'type::signal_quality': u4
+                'type::noise': u4
+                'type::data_rate': u4
+                'type::is_tx': u4
+                'type::frame_length': u4
+        instances:
+          is_supplied:
+            value: is_not_supplied == 0
+        types:
+          header:
+            # all of these are wild guesses
+            seq:
+              - id: header_size
+                type: b4
+              - id: mandatory_part_offset
+                type: b4
+              - id: placeholder
+                type: placeholder
+                size: header_size - 1
+            instances:
+              mandatory_part_offset_from_first_byte:
+                value: mandatory_part_offset - 1
+              mandatory_part:
+                pos: mandatory_part_offset_from_first_byte / 2
+                type: mandatory_part(mandatory_part_skip_nibble)
+                io: placeholder._io
+              mandatory_part_skip_nibble:
+                value: mandatory_part_offset_from_first_byte % 2 == 1
+              type:
+                value: mandatory_part.type
```

Comment:
Oh my God, this is so much over-engineered! Let me remind you that according to the [specification](https://www.tcpdump.org/linktypes/LINKTYPE_IEEE802_11_PRISM.html) from `doc-ref` (yeah, I know it's bad to rely on a single source, but I don't know any others), your `header_size` attribute is always equal to **4** and the only valid values for `mandatory_part_offset` are 1 and 4. So your spec permits lots of combinations that are likely to be incompatible with any other parser implementations. This flexibility may seem OK and basically in the spirit of the [Postel's law](https://en.wikipedia.org/wiki/Robustness_principle) (_"be liberal in what you accept from others"_), but IMHO here it does more harm than good. Consider that your format doesn't have any safeguards against invalid packets. This doesn't hurt by itself, but it sucks when you can't even check